test/functional/CacheManagement/src/tests/sharedclasses/options/TestSharedCacheJavaAPI.java

```diff
@@ -106,9 +168,28 @@ public static void main(String args[]) {
 		    	fail("SharedClassUtilities.getSharedCacheInfo failed: no cache found");
 		    }
 		    newCacheCount = cacheList.size();
-		    if ((newCacheCount == -1) || (newCacheCount != (oldCacheCount + persistentCount + nonpersistentCount + snapshotCount))) {
-		    	fail("SharedClassUtilities.getSharedCacheInfo failed: Invalid number of cache found\t" +
-		    			"expected: " + (oldCacheCount + persistentCount + nonpersistentCount + snapshotCount) + "\tfound: " + newCacheCount);
+		    
+		    if (dirGroupAccess != null) {
+		    	cacheGroupAccessList = SharedClassUtilities.getSharedCacheInfo(dirGroupAccess, SharedClassUtilities.NO_FLAGS, false);
+		    	if (cacheGroupAccessList == null) {
```

Comment:
Do you want to check cacheGroupAccessList .size() is 0 as well ?

Reply:
It could be 0. And any wrong case should be covered by the check

``` 
(newCacheCount != 
		    			(oldCacheCount + persistentGroupAccessCount + nonpersistentGroupAccessCount +
		    					persistentCount + nonpersistentCount + snapshotCount)))
``` 

Not sure if I understood correctly.

Reply:
OK. Then probably it is better to include each value of oldCacheCount, persistentGroupAccessCount, nonpersistentGroupAccessCount, persistentCount, nonpersistentCount, snapshotCount in the error message if the sum does not equals to newCacheCount.

Reply:
updated.